File: src/main/java/com/heybys/oddments/fooddelivery/persistence/user/UserJpaRepository.java
package com.heybys.oddments.fooddelivery.persistence.user;

import java.util.Optional;

import org.springframework.data.jpa.repository.JpaRepository;

import com.heybys.oddments.fooddelivery.domain.user.User;
import com.heybys.oddments.fooddelivery.domain.user.UserId;

interface UserJpaRepository extends JpaRepository<User, UserId> {
    Optional<User> findByEmail(String email);
}

File: src/main/kotlin/com/heybys/oddments/fooddelivery/persistence/user/UserJpaRepository.kt
package com.heybys.oddments.fooddelivery.persistence.user

import com.heybys.oddments.fooddelivery.domain.user.User
import com.heybys.oddments.fooddelivery.domain.user.UserId
import org.springframework.data.jpa.repository.JpaRepository
import java.util.Optional

internal interface UserJpaRepository : JpaRepository<User, UserId> {
    fun findByEmail(email: String): Optional<User>
}
